Design/Instrument-2.0/Detector_Scan_NeXus_Saving.md

```diff
@@ -0,0 +1,129 @@
+
+
+## Implementation Ideas
+
+### New NeXus Instrument Format
+
+_Based on discussion with Matt Jones_
+
+Work has been done for the ESS for a new instrument format in the NeXus files. The approach in the new format, effectively a replacement for the old XML IDF, contains detectors (components in Mantid terminology) with pixels, for example a detector tube or bank with sub-pixels.
+
+See the example for the NeXus hierarchy below. For a component with pixels in 3-dimensions the detector numbers are given, and then the offsets are given in x, y and z. Not all of the offsets are required, for example a tube might only require the `y_pixel_offset` entry.
+
+```
+raw_data_1
+├───instrument
+│   ├───detector_1
+│   │   ├───detector_number (size is number of detectors)
+│   │   ├───x_pixel_offset (size is number of detectors)
+│   │   ├───y_pixel_offset (size is number of detectors)
+│   │   ├───z_pixel_offset (size is number of detectors)
+│   │   ├───...
+│   ├───detector_2
+│   ├───...
+├───sample
+```
+
+The suggested method for dealing with detector scans is to add entries as `NXTransformation`s to the detector entry. This is shown in the example hierarchy below. For each position and rotation axis a separate entry is required within the detector, such as `position_scan_x`, `position_scan_y`, `position_scan_z`, `rotation_scan_x`, `rotation_scan_y` and `rotation_scan_z`.
+
+```
+raw_data_1
+├───instrument
+│   ├───detector_1
+│   │   ├───detector_number
+│   │   ├───x_pixel_offset
+│   │   ├───y_pixel_offset
+│   │   ├───z_pixel_offset
+│   │   ├───position_scan_x (attribute 'vector' for transformation, e.g. (1., 0., 0.))
+│   │   │   ├───scan_log
+│   │   │       ├───time (contains the time for each entry in value)
+│   │   │       ├───value (single number per time entry, describing the distance along 'vector')
+│   │   ├───position_scan_...
+│   ├───detector_2
+│   ├───...
+├───sample
+```
+
+An open question here, especially when considering conversion from the old IDFs, is how to define the component used for `detector_X`. Technically this could be anything from the whole instrument to individual pixels, but logically it should be something like a tube. The only requirement is that a detector has pixels with a fixed offset at all times.
+
+#### Differences Between NeXus and Mantid Approach
+
+In the `DetectorInfo` representation in Mantid each position, rotation and mask flag of each pixel at each time index is stored separately. The scans are done on a pixel-by-pixel basis, and the components can not currently do any scanning. Different pixels are allowed to be given different scan intervals even within a component.
+
+The NeXus instrument proposal would not support scanning at the pixel level. The offset entries would be the same for each time index, with the position entries giving the scan for a component.
```

Comment:
Can you clarify what you mean when you say the "offset entries would be the same for each time index"

Reply:
I think it basically means that you cannot "scale" a bank via a scan, i.e., the pixel gap is constant in time.

Reply:
Thanks, that makes sense.